app/assets/javascripts/covid19/src/webapp/components/structures-table/NMRDialog.tsx

```diff
@@ -0,0 +1,304 @@
+import _ from "lodash";
+import React from "react";
+import styled from "styled-components";
+import {
+    Button,
+    Table,
+    TableBody,
+    TableCell,
+    TableContainer,
+    TableHead,
+    TablePagination,
+    TableRow,
+    Typography,
+    LinearProgress,
+    CircularProgress,
+    Paper,
+} from "@material-ui/core";
+import { GetApp, Stop as StopIcon } from "@material-ui/icons";
+import { Dialog } from "./Dialog";
+import { NMROptions, SetNMRPagination } from "./Columns";
+import { NSPTarget, getValidationSource } from "../../../domain/entities/Covid19Info";
+import { NMRPagination } from "../../../domain/repositories/EntitiesRepository";
+import { useBooleanState } from "../../hooks/useBoolean";
+import { useAppContext } from "../../contexts/app-context";
+import { NoBulletListItem as ListItem } from "./IDRDialog";
+import i18n from "../../../utils/i18n";
+import { Cancel } from "../../../data/utils/future";
+
+export interface NMRDialogProps {
+    onClose(): void;
+    nmrOptions: NMROptions;
+    open: boolean;
+}
+
+export const NMRDialog: React.FC<NMRDialogProps> = React.memo(props => {
+    const { onClose, open } = props;
+    const { target, error, pagination, setPagination, loading } = props.nmrOptions;
+    const { compositionRoot, sources } = useAppContext();
+    const [isExporting, { enable: showExporting, disable: hideExporting }] = useBooleanState(false);
+
+    const nmrSource = React.useMemo(() => getValidationSource(sources, "NMR"), [sources]);
+    const nmrMethod = nmrSource?.methods[0];
+
+    const title = React.useMemo(
+        () => `${i18n.t("NMR-based fragment screening on")}: ${target?.name ?? ""}`,
+        [target]
+    );
```

Comment:
No need to change, but no need to useMemo for i18n strings (it will be probably be slower than just calling i18n.t)